wgengine/magicsock/derp.go

```diff
@@ -542,6 +542,17 @@ func (c *Conn) runDerpReader(ctx context.Context, derpFakeAddr netip.AddrPort, d
 					c.addDerpPeerRoute(res.src, regionID, dc)
 				}
 			}
+			select {
+			case <-ctx.Done():
+				return
+			case c.derpRecvCh <- res:
```

Comment:
Now that we're not using `res` outside of this `case` branch, and since it's a value type already, we may be able to move the declaration into the case branch to make sure that nothing else uses it?

Reply:
I'm hesitant to move it for performance reasons around assignment of the closure. Ideally the toolchain would make this a non-issue, but keeping it as-is for now. The docs on the type make it clear that it should only be used for `derp.ReceivedPacket`.

Reply:
SGTM; thanks 👍